### ISSUE TITLE
Cannot access pivot table settings (sidebar) from a base64 encoded ad-hoc URL

### ISSUE THREAD

**Issue by nemanjaglumac** [metabase/metabase#14699]
**Describe the bug**
When visiting ad-hoc question through the JWT-encoded link, it is impossible to open a pivot table settings (sidebar). You can click on the "Settings", but sidebar doesn't open.

**To Reproduce**
1. Visit `{rootDir}/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJxdWVyeSIsInF1ZXJ5Ijp7InNvdXJjZS10YWJsZSI6MiwiYWdncmVnYXRpb24iOltbImNvdW50Il1dLCJicmVha291dCI6W1siZmstPiIsWyJmaWVsZC1pZCIsMTFdLFsiZmllbGQtaWQiLDI2XV0sWyJmay0-IixbImZpZWxkLWlkIiwxM10sWyJmaWVsZC1pZCIsNF1dXX0sImRhdGFiYXNlIjoxfSwiZGlzcGxheSI6InBpdm90IiwiZGlzcGxheUlzTG9ja2VkIjp0cnVlfQ==`
2. Click on "Settings"
3. Sidebar doesn't open
4. JS Console error
```
Uncaught TypeError: Cannot read property 'virtual_card' of undefined
    at ChartSettings.render (ChartSettings.jsx:253)
    at ChartSettings.<anonymous> (makeAssimilatePrototype.js:15)
    at ReactCompositeComponent.js:793
    at measureLifeCyclePerf (ReactCompositeComponent.js:73)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (ReactCompositeComponent.js:792)
    at ReactCompositeComponentWrapper._renderValidatedComponent (ReactCompositeComponent.js:819)
    at ReactCompositeComponentWrapper.performInitialMount (ReactCompositeComponent.js:359)
    at ReactCompositeComponentWrapper.mountComponent (ReactCompositeComponent.js:255)
    at Object.mountComponent (ReactReconciler.js:43)
    at ReactDOMComponent.mountChildren (ReactMultiChild.js:234)
```

**Expected behavior**
Clicking on a pivot table "Settings" button should open sidebar.

**Screenshots**
![ ](https://p58.f1.n0.cdn.getcloudapp.com/items/2NuEBNmn/b189ebde-2ab1-419e-9d9c-195d5625c2da.gif?source=viewer&v=34419a1794b4b7823d04f40aab69c53a)

**Information about your Metabase Installation:**
- Latest commit on `release-x.38.x` branch (at the time of writing, it's db8a8e1f5d5704da27c501dec2f05796b7c32e44)

**Severity**
Not sure how many people use JWT encoded links to access certain question. So far, the biggest annoyance is that our Cypress tests started failing, because we use `visitQuestionAdhoc()` function a lot.

**Additional context**
- PR  #14698 has been opened to temporarily skip failing tests and enable people to work on `release-x.38.x` branch.
- Examples of failing tests:
    - https://app.circleci.com/pipelines/github/metabase/metabase/12715/workflows/b7e78988-0cf9-4574-aae3-c3270d769949/jobs/478609
    - https://app.circleci.com/pipelines/github/metabase/metabase/12715/workflows/b7e78988-0cf9-4574-aae3-c3270d769949/jobs/478603

**Comment by flamber** [comment]
Got a little confused. JWT is not the same as the base64 encoded question query string.
I'm unsure what the problem is, but I'm having a hard time reproducing - db8a8e1 or rc4 - and I cannot use the example string, since my sample data is not synced to the same IDs as yours.

When I decode your example I get:
`{"dataset_query":{"type":"query","query":{"source-table":2,"aggregation":[["count"]],"breakout":[["fk->",["field-id",11],["field-id",26]],["fk->",["field-id",13],["field-id",4]]]},"database":1},"display":"pivot","displayIsLocked":true}`

But when I try to create a similar query (Custom > Orders Count by User.Source and Product Category > change viz to Pivot Table), I get the following:
`{"dataset_query":{"database":4,"query":{"source-table":11,"aggregation":[["count"]],"breakout":[["fk->",["field-id",80],["field-id",95]],["fk->",["field-id",82],["field-id",107]]]},"type":"query"},"display":"pivot","displayIsLocked":true,"visualization_settings":{"pivot_table.column_split":{"rows":[["fk->",["field-id",80],["field-id",95]]],"columns":[["fk->",["field-id",82],["field-id",107]]],"values":[["aggregation",0]]}}}`

That's a lot of extra information added, when I manually select Pivot Table as the visualization compared to your query string.

**Comment by nemanjaglumac** [comment]
> Got a little confused. JWT is not the same as the base64 encoded question query string.

My bad. I got so used to the fact that we're using JWT for base64 encoding that I mix the two. Since we don't need or use any `secret` for the signature, this is just regular base64 encoded string.

This is the code Paul made for this purpose:
```javascript
export function adhocQuestionHash(question) {
  if (question.display) {
    // without "locking" the display, the QB will run its picking logic and override the setting
    question = Object.assign({}, question, { displayIsLocked: true });
  }
  return btoa(unescape(encodeURIComponent(JSON.stringify(question))));
}

export function visitQuestionAdhoc(question) {
  cy.visit("/question#" + adhocQuestionHash(question));
}
```

Hope it helps with the reproduction.

**Comment by paulrosenzweig** [comment]
Closed by #14704
